src/hotspot/share/opto/phasetype.hpp

```diff
@@ -75,6 +75,12 @@
   flags(ITER_GVN_AFTER_ELIMINATION,     "Iter GVN after Eliminating Allocations and Locks") \
   flags(BEFORE_PRE_MAIN_POST,           "Before Pre/Main/Post Loops") \
   flags(AFTER_PRE_MAIN_POST,            "After Pre/Main/Post Loops") \
+  flags(BEFORE_POST_LOOP,               "Before Post Loop") \
+  flags(AFTER_POST_LOOP,                "After Post Loop") \
+  flags(BEFORE_REMOVE_EMPTY_LOOP,       "Before Remove Empty Loop") \
+  flags(AFTER_REMOVE_EMPTY_LOOP,        "After Remove Empty Loop") \
+  flags(BEFORE_ONE_ITERATION_LOOP,      "Before Replacing One Iteration Loop") \
+  flags(AFTER_ONE_ITERATION_LOOP,       "After Replacing One Iteration Loop") \
```

Comment:
Very much a nit, but I think this should be "One-Iteration Loop". Or, is it in fact one _iteration loop_ (as it reads now)? Looking at the code, I think it is the former. @chhagedorn can maybe clarify?

This is not specific to your changeset, but also appears in existing source code comments. Maybe a good opportunity to clean this up everywhere?

Also, maybe "Replacing" should be "Replace"? Seems to better fit the style used for other phase names.

Reply:
One-Iteration loop sounds better indeed. I also agree with the other suggestions.

Something else I've noticed is that we could also benefit when we add dumps for `duplicate_loop_backedge()` which creates a new loop node (i.e. could be seen as "major modification"). I just looked into recently and found myself adding dumps there manually for debugging. I guess since this is a dump adding RFE, we could also add that one. What do you think? But then we would need to update the PR title to something like "add various new graph dumps during loop opts".